docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+- adding an `Image` component to your `Cat` component
+- setting `Image`‘s `source` via `Cat`‘s `props` with `{props.picURI}`
+- then passing `picURI` values in each `Cat`
+
+```SnackPlayer name=Custom%20Pictures
+import React from 'react';
+import { Text, View, Image } from 'react-native';
+
+function Cat(props) {
+  return (
+    <View>
+      <Image source={props.picURI} style={{width: 200, height: 200}}/>
+      <Text>Hello, I am {props.name}!</Text>
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <View>
+      <Cat name="Maru" picURI="https://placekitten.com/g/200/200" />
+      <Cat name="Jellylorum" picURI="https://placekitten.com/g/200/200" />
+      <Cat name="Spot" picURI="https://placekitten.com/g/200/200" />
+    </View>
+  );
+}
+```
+
+You can build many things with `props` and the Core Components [`Text`](text), [`Image`](image), and [`View`](view)! But to build something interactive, you’ll need `state`.
+
+## State
+
+Props are set by the parent component and can only be changed by the parent component passing down new values. But for individual components to change according to human interactions, they need to be able to accept and store their own data. Fortunately, React components can have their own `state` you can get and set data from.
+
+<div class="toggler">
+  <ul role="tablist" class="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTabs('syntax', 'functional')">
+      State with Function Components
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTabs('syntax', 'classical')">
+      State with Class Components
+    </li>
+  </ul>
+</div>
+
+<block class="functional syntax" />
+
+You can add state to a component by calling [React’s `useState` Hook](https://reactjs.org/docs/hooks-state.html).
+
+> A Hook is a kind of function that lets you “hook into” React features. For example, `useState` is a Hook that lets you add state to function components. You can learn more about [other kinds of Hooks in the React documentation.](https://reactjs.org/docs/hooks-intro.html)
+
+You can declare a component’s state by calling `useState` inside its function like this:
+
+`const [currentValue, setNewValue] = useState("Initial value.");`
+
+You can use `useState` to track any kind of data, including numbers:
+
+`const [quantity, pickQuantity] = useState(0);`
+
+Calling `useState` does two things:
+
+- it creates a “state variable” with an initial value
+- it creates a function to set that state variable’s value
+
+Let’s break `[a, b] = useState(c)` down:
+
+- `a` is the “state variable”—it refers to the current value of the component’s state
+- `b` is the name of the function you use to update the state variable
+- `c` is the value you initialize the state variable with
+
```

Comment:
I think at this point I'd still wonder: why do I need a "state variable"?

Maybe you could say: 

>Normally, when you exit a function, its variables disappear. But we want our components to remember some data —for example, the user input. This is where state variables come in. They let components remember things! Here's how to declare a state variable.

Reply:
In general, Hooks are kind of a generalization of this. They let things "stick around".

Reply:
I think my response above https://github.com/facebook/react-native-website/pull/1535#discussion_r367682836 should clear this up?